src/python/pants/backend/python/subsystems/pex_build_util.py

```diff
@@ -55,15 +55,27 @@ def has_python_requirements(tgt):
   return isinstance(tgt, PythonRequirementLibrary)
 
 
-def can_have_python_platform(tgt):
-  return isinstance(tgt, (PythonBinary, PythonDistribution))
+def may_add_python_platform_to_resolve(tgt):
+  return isinstance(tgt, (PythonBinary, PythonDistribution, PythonTests))
 
 
 def targets_by_platform(targets, python_setup):
   d = defaultdict(OrderedSet)
   for target in targets:
-    if can_have_python_platform(target):
-      for platform in target.platforms if target.platforms else python_setup.platforms:
+    if may_add_python_platform_to_resolve(target):
+      # PythonTests doesn't have a platforms property, and should (currently) always run on the
+      # default configured platform, so we getattr and fall back to the configured value one if
+      # the target doesn't have a platforms property.
+      #
+      # If we don't do this, the default configured platform may not be resolved (specifically if a
+      # PythonBinary is also built at the same time, which explicitly defines platforms not
```

Comment:
What about PythonDistribution? I would expect the same effect.

Reply:
`PythonDistribution` _does_ explicitly have a `platforms` property, and so is already covered here (because it's in the list of types checked in `may_add_python_platform_to_resolve`)

Reply:
Removed comment :)